### PR TITLE
feat(core): Support community packages in scaling-mode 

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This adds support for community nodes in queue-mode.

## Related Linear tickets, Github issues, and Community forum posts
https://linear.app/n8n/issue/ENG-135
https://community.n8n.io/t/install-community-nodes-in-queue-mode-on-n8n-1-x-x/29261
https://community.n8n.io/t/community-node-installation-with-n8n-running-in-queue-mode/18081

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included

### CODE REVIEW

File: packages/cli/src/services/orchestration/main/handleCommandMessageMain.ts [hunk 1]
```diff
@@ -77,6 +78,20 @@ export async function handleCommandMessageMain(messageString: string) {
...
+					return message;
+				}
+				const { packageName, packageVersion } = message.payload;
+				const communityPackagesService = Container.get(CommunityPackagesService);
+				if (message.command === 'community-package-uninstall') {
+					await communityPackagesService.removeNpmPackage(packageName);
+				} else {
+					await communityPackagesService.installOrUpdateNpmPackage(packageName, packageVersion);
+				}
+				break;
```
Comment (ivov) [inline packages/cli/src/services/orchestration/main/handleCommandMessageMain.ts:94]: So this should work for multi-main as well right?

**Review by ivov — COMMENTED** [review]
Nice work :)

I'll come back later to test manually.

- Can we please edit title to use `feat:` and `scaling mode` instead?
- Can we please add a couple of forum threads to notify for future searchers?

**Review by valya — DISMISSED** [review]

File: packages/cli/src/commands/webhook.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ export class Webhook extends BaseCommand {
 
 	protected server = Container.get(WebhookServer);
 
+	override needsCommunityNodes = true;
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/commands/webhook.ts:25]: I think this is needed for this one check https://github.com/n8n-io/n8n/blob/master/packages/cli/src/WebhookHelpers.ts#L235

File: packages/cli/src/services/orchestration/main/handleCommandMessageMain.ts (hunk 1 above)
Comment (netroy) [reply packages/cli/src/services/orchestration/main/handleCommandMessageMain.ts:94]: I believe so. but I haven't tested this bit.

File: packages/cli/src/commands/start.ts [hunk 3]
```diff
@@ -176,6 +177,22 @@ export class Start extends BaseCommand {
 			this.logger.debug(`Queue mode id: ${this.queueModeId}`);
 		}
 
+		const { flags } = await this.parse(Start);
+		const { communityPackages } = this.globalConfig.nodes;
+		// cli flag overrides the config env variable
+		if (flags.reinstallMissingPackages) {
+			if (communityPackages.enabled) {
+				this.logger.warn(
+					'`--reinstallMissingPackages` is deprecated: Please use the env variable `N8N_REINSTALL_MISSING_PACKAGES` instead',
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/commands/start.ts:187]: https://github.com/n8n-io/n8n-docs/pull/2304

**Review by ivov — DISMISSED** [review]
Thanks for this! This is going to unblock many users.

File: packages/cli/src/commands/audit.ts [hunk 4]
```diff
@@ -30,6 +30,8 @@ export class SecurityAudit extends BaseCommand {
 		}),
 	};
 
+	override needsCommunityPackages = true;
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/commands/audit.ts:33]: I assumed that `NodesRiskReporter` needed that. but looks like it only looks at the DB entries. I'll revert this.

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)

**Pull request merged**